pkg/apimachinery/utils/meta.go

```diff
@@ -821,3 +829,97 @@ func (m *grafanaMetaAccessor) SetSourceProperties(v SourceProperties) {
 
 	m.obj.SetAnnotations(annot)
 }
+
+// GetSecureValues implements GrafanaMetaAccessor.
+func (m *grafanaMetaAccessor) GetSecureValues() (vals common.InlineSecureValues, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("error reading spec")
+		}
+	}()
+
+	var property any // may be map or struct
+
+	f := m.r.FieldByName("Secure")
+	if f.IsValid() {
+		property = f.Interface()
+	} else {
+		// Unstructured
+		u, ok := m.raw.(*unstructured.Unstructured)
+		if ok {
+			property = u.Object["secure"]
+		}
+	}
+
+	// Not found (and no error)
+	if property == nil {
+		return nil, nil
+	}
+
+	// Try directly casting the property
+	vals, ok := property.(common.InlineSecureValues)
+	if ok {
+		return vals, nil
+	}
+
+	// Generic map
+	u, ok := property.(map[string]any)
```

Comment:
I'm just a bit confused when can this case happen? Shouldn't everyone be using `common.InlineSecureValues`? 🤔